src/clusto/drivers/base/resourcemanager.py

```diff
@@ -195,12 +200,20 @@ def deallocate(self, thing, resource=(), number=True):
         try:
             if resource is ():                      
                 for res in self.resources(thing):
+                    res_mgr = self.get_resource_manager(res)
+                    if res_mgr != self:
+                        continue
                     thing.del_attrs(self._attr_name, number=res.number)
 
             elif resource and not self.available(resource, number):
                 resource, number = self.ensure_type(resource, number)
+                try:
+                    number = self.get_resource_number(thing, resource)
+                except:
```

Comment:
Blanket `except` clauses are generally dangerous in python. You should probably enumerate the exceptions you are actually concerned about instead.